src/ethereum/utils.py

```diff
@@ -59,3 +76,354 @@ def ceil32(value: Uint) -> Uint:
+    """
+    return Bytes8(bytes.fromhex(remove_hex_prefix(hex_string).rjust(16, "0")))
+
+
+def hex_to_bytes32(hex_string: str) -> Bytes32:
+    """
+    Convert hex string to 32 bytes.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to 32 bytes.
+
+    Returns
+    -------
+    32_byte_stream : `bytes`
+        32-byte stream corresponding to the given hexadecimal string.
+    """
+    return Bytes32(bytes.fromhex(remove_hex_prefix(hex_string).rjust(64, "0")))
+
+
+def hex_to_hash(hex_string: str) -> Hash32:
+    """
+    Convert hex string to hash32 (32 bytes).
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to hash32.
+
+    Returns
+    -------
+    hash : `Hash32`
+        32-byte stream obtained from the given hexadecimal string.
+    """
+    return Hash32(bytes.fromhex(remove_hex_prefix(hex_string)))
+
+
+def hex_to_root(hex_string: str) -> Root:
+    """
+    Convert hex string to trie root.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to trie root.
+
+    Returns
+    -------
+    root : `Root`
+        Trie root obtained from the given hexadecimal string.
+    """
+    return Root(bytes.fromhex(remove_hex_prefix(hex_string)))
+
+
+def hex_to_bloom(hex_string: str) -> Bloom:
+    """
+    Convert hex string to bloom.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to bloom.
+
+    Returns
+    -------
+    bloom : `Bloom`
+        Bloom obtained from the given hexadecimal string.
+    """
+    return Bloom(bytes.fromhex(remove_hex_prefix(hex_string)))
+
+
+def hex_to_address(hex_string: str) -> Address:
+    """
+    Convert hex string to Address (20 bytes).
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to Address.
+
+    Returns
+    -------
+    address : `Address`
+        The address obtained from the given hexadecimal string.
+    """
+    return Address(bytes.fromhex(remove_hex_prefix(hex_string).rjust(40, "0")))
+
+
+def hex_to_uint(hex_string: str) -> Uint:
+    """
+    Convert hex string to Uint.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to Uint.
+
+    Returns
+    -------
+    converted : `Uint`
+        The unsigned integer obtained from the given hexadecimal string.
+    """
+    return Uint(int(remove_hex_prefix(hex_string), 16))
+
+
+def hex_to_u256(hex_string: str) -> U256:
+    """
+    Convert hex string to U256.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be converted to U256.
+
+    Returns
+    -------
+    converted : `U256`
+        The U256 integer obtained from the given hexadecimal string.
+    """
+    return U256(int(remove_hex_prefix(hex_string), 16))
+
+
+def rlp_hash(data: Any) -> Hash32:
```

Comment:
```suggestion
def rlp_hash(data: rlp.RLP) -> Hash32:
```

(I know mypy treats them as the same, but this is technically correct.)